src/enrichment_tables/file.rs

```diff
@@ -114,16 +176,34 @@ impl File {
         self.headers.iter().position(|header| header == col)
     }
 
-    fn row_equals(&self, condition: &[Condition], row: &[String]) -> bool {
+    /// Currently all matches are case insensitive.
+    /// TODO We want to add a configuration option to allow for case sensitive searches. This will
+    /// allow for more performant matches.
+    fn row_equals(&self, condition: &[Condition], row: &[Value]) -> bool {
         condition.iter().all(|condition| match condition {
             Condition::Equals { field, value } => match self.column_index(field) {
                 None => false,
-                Some(idx) => row[idx].to_lowercase() == value.to_lowercase(),
+                Some(idx) => match (&row[idx], value) {
+                    (Value::Bytes(bytes1), Value::Bytes(bytes2)) => {
+                        match (std::str::from_utf8(bytes1), std::str::from_utf8(bytes2)) {
+                            (Ok(s1), Ok(s2)) => s1.to_lowercase() == s2.to_lowercase(),
+                            _ => bytes1 == bytes2,
+                        }
```

Comment:
I guess you could a case to bail out early:

```suggestion
                        match (std::str::from_utf8(bytes1), std::str::from_utf8(bytes2)) {
                            (Ok(s1), Ok(s2)) => s1.to_lowercase() == s2.to_lowercase(),
                            (Err(_), Err(_)) => bytes1 == bytes2,
                            _ => false,
                        }
```